### PR TITLE
8296190: TestMD5Intrinsics and TestMD5MultiBlockIntrinsics don't test the intrinsics

### PR DESCRIPTION
**Pull request by yftsai**
Increase WARMUP_ITERATIONS to get MD5 hash from the intrinsic.

Both TestMD5Intrinsics and TestMD5MultiBlockIntrinsics are run with options "-XX:CompileThreshold=500 -XX:Tier4InvocationThreshold=500", so the current value is too low for the intrinsic to be executed.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8296190](https://bugs.openjdk.org/browse/JDK-8296190): TestMD5Intrinsics and TestMD5MultiBlockIntrinsics don't test the intrinsics

### Reviewers
 * [Evgeny Astigeevich](https://openjdk.org/census#eastigeevich) (@eastig - Committer)
 * [Paul Hohensee](https://openjdk.org/census#phh) (@phohensee - **Reviewer**)
 * [Volker Simonis](https://openjdk.org/census#simonis) (@simonis - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10954/head:pull/10954` \
`$ git checkout pull/10954`

Update a local copy of the PR: \
`$ git checkout pull/10954` \
`$ git pull https://git.openjdk.org/jdk pull/10954/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10954`

View PR using the GUI difftool: \
`$ git pr show -t 10954`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10954.diff">https://git.openjdk.org/jdk/pull/10954.diff</a>

</details>

### CODE REVIEW

**Review by eastig — APPROVED** [review]
Lgtm

**Review by phohensee — APPROVED** [review]
Lgtm, and trivial.

**Comment by simonis** [comment]
I'm a little surprised that this isn't detected. The tests you mention have a verification step (i.e. `compiler.testlibrary.intrinsics.Verifier`) which should detect this. Can you please check why that is not the case and if this also affects other intrinsic tests?

**Review by simonis — CHANGES_REQUESTED** [review]
I'm a little surprised that this isn't detected. The tests you mention have a verification step (i.e. compiler.testlibrary.intrinsics.Verifier) which should detect this. Can you please check why that is not the case and if this also affects other intrinsic tests?

Also, it is a little unfortunate, that the new warmup value still implicitly depends on the -XX:CompileThreshold=500 setting in the actual test. It would be more stable to get that value explicitly (e.g. by using the WhiteBox API) and use a strictly bigger warmup value.

Finally, I think the actual tests should be run with -Xbatch to be sure that the corresponding methods have really been compiled.

**Comment by phohensee** [comment]
Hi, Volker. Perhaps we could let this patch go in to fix the immediate problem, and have Yifan file an issue for a definitive fix?

**Comment by yftsai** [comment]
The verifier detects the intrinsic compiled/inlined somewhere, but it doesn't get executed. In other words, the verifier alone didn't guarantee the intrinsic tested.

WARMUP_ITERATIONS is now set strictly larger than Tier4InvocationThreshold.

Both TestMD5Intrinsics and TestMD5MultiBlockIntrinsics are run with `-Xbatch`.

File: test/hotspot/jtreg/compiler/intrinsics/sha/sanity/DigestSanityTestBase.java [hunk 1]
```diff
@@ -54,7 +54,7 @@ public class DigestSanityTestBase {
     private static final int MSG_SIZE = 1024;
     private static final int OFFSET = 0;
     private static final int ITERATIONS = 10000;
-    private static final int WARMUP_ITERATIONS = 550;
+    private static final int WARMUP_ITERATIONS = WHITE_BOX.getIntxVMFlag("Tier4InvocationThreshold").intValue() + 50;
```
Comment (eastig) [inline test/hotspot/jtreg/compiler/intrinsics/sha/sanity/DigestSanityTestBase.java:57]: If the flag `Tier4InvocationThreshold` is not set, what will happen?

Comment (yftsai) [reply test/hotspot/jtreg/compiler/intrinsics/sha/sanity/DigestSanityTestBase.java:57]: The default value, 5000, would be returned.

**Review by simonis — APPROVED** [review]
Hi Yi-Fan,

Thanks for doing the additional changes and for confirming that the test run with `-Xbatch`.

Looks good now.

Best regards,
Volker

**Review by eastig — APPROVED** [review]

**Review by phohensee — APPROVED** [review]
Lgtm.

**Pull request closed without merging**